pinot-common/src/main/java/org/apache/pinot/common/metadata/ZKMetadataProvider.java

```diff
@@ -144,34 +141,25 @@ public static void removeResourceConfigFromPropertyStore(ZkHelixPropertyStore<ZN
     }
   }
 
-  public static boolean setOfflineSegmentZKMetadata(ZkHelixPropertyStore<ZNRecord> propertyStore,
-      String offlineTableName, OfflineSegmentZKMetadata offlineSegmentZKMetadata, int expectedVersion) {
+  public static boolean setSegmentZKMetadata(ZkHelixPropertyStore<ZNRecord> propertyStore, String tableNameWithType,
+      SegmentZKMetadata segmentZKMetadata, int expectedVersion) {
     // NOTE: Helix will throw ZkBadVersionException if version does not match
     try {
       return propertyStore
-          .set(constructPropertyStorePathForSegment(offlineTableName, offlineSegmentZKMetadata.getSegmentName()),
-              offlineSegmentZKMetadata.toZNRecord(), expectedVersion, AccessOption.PERSISTENT);
+          .set(constructPropertyStorePathForSegment(tableNameWithType, segmentZKMetadata.getSegmentName()),
+              segmentZKMetadata.toZNRecord(), expectedVersion, AccessOption.PERSISTENT);
     } catch (ZkBadVersionException e) {
       return false;
     }
   }
 
-  public static boolean setOfflineSegmentZKMetadata(ZkHelixPropertyStore<ZNRecord> propertyStore,
-      String offlineTableName, OfflineSegmentZKMetadata offlineSegmentZKMetadata) {
-    return propertyStore
-        .set(constructPropertyStorePathForSegment(offlineTableName, offlineSegmentZKMetadata.getSegmentName()),
-            offlineSegmentZKMetadata.toZNRecord(), AccessOption.PERSISTENT);
-  }
-
-  public static boolean setRealtimeSegmentZKMetadata(ZkHelixPropertyStore<ZNRecord> propertyStore,
-      String realtimeTableName, RealtimeSegmentZKMetadata realtimeSegmentZKMetadata) {
-    return propertyStore
-        .set(constructPropertyStorePathForSegment(realtimeTableName, realtimeSegmentZKMetadata.getSegmentName()),
-            realtimeSegmentZKMetadata.toZNRecord(), AccessOption.PERSISTENT);
+  public static boolean setSegmentZKMetadata(ZkHelixPropertyStore<ZNRecord> propertyStore, String tableNameWithType,
+      SegmentZKMetadata segmentZKMetadata) {
+    return setSegmentZKMetadata(propertyStore, tableNameWithType, segmentZKMetadata, -1);
   }
 
   @Nullable
-  public static ZNRecord getZnRecord(@Nonnull ZkHelixPropertyStore<ZNRecord> propertyStore, @Nonnull String path) {
+  public static ZNRecord getZnRecord(ZkHelixPropertyStore<ZNRecord> propertyStore, String path) {
```

Comment:
why removing all @Nonnull annotation?

Reply:
`Nonnull` and `Nullable` looks quite similar, so we usually only keep `Nullable` and assume the variables are `Nonnull` by default (google convention)

Reply:
ic. I see some methods mark one param as Nullable and leave others without annotation. In this case, does compile assume the other Nonnull or Nullable?

Reply:
This annotation is mostly for readability purpose, and will be ignored by the compiler. IDE will perform some simple check and warn when invoking methods on `nullable` values.